src/components/CentralContainer/Sidebar/SidebarTabs/SearchTabPanel/index.tsx

```diff
@@ -0,0 +1,128 @@
+    const [queryOptions, setQueryOptions] = useState<QUERY_OPTION[]>([]);
+    const handleQueryInputChange = (ev: React.ChangeEvent<HTMLTextAreaElement>) => {
+        const isCaseSensitive = queryOptions.includes(QUERY_OPTION.IS_CASE_SENSITIVE);
+        const isRegex = queryOptions.includes(QUERY_OPTION.IS_REGEX);
+        startQuery(ev.target.value, isRegex, isCaseSensitive);
+    };
+
+    return (
+        <CustomTabPanel
+            tabName={TAB_NAME.SEARCH}
+            title={TAB_DISPLAY_NAMES[TAB_NAME.SEARCH]}
+            titleButtons={
+                <TitleButton onClick={() => { setIsAllExpanded((v) => !v); }}>
+                    {isAllExpanded ?
+                        <UnfoldLessIcon/> :
+                        <UnfoldMoreIcon/>}
+                </TitleButton>
+            }
+        >
+            <Box className={"search-tab-container"}>
+                <div>
+                    <Textarea
+                        className={"query-input-box"}
+                        disabled={isDisabled(uiState, UI_ELEMENT.QUERY_INPUT_BOX)}
+                        maxRows={7}
+                        placeholder={"Search"}
+                        size={"sm"}
+                        endDecorator={
+                            <ToggleButtonGroup
+                                size={"sm"}
+                                spacing={0.25}
+                                value={queryOptions}
+                                variant={"plain"}
+                                onChange={(_, newValue) => {
+                                    setQueryOptions(newValue);
+                                }}
+                            >
+                                <IconButton
+                                    className={"query-option"}
+                                    value={QUERY_OPTION.IS_CASE_SENSITIVE}
+                                >
+                                    Aa
+                                </IconButton>
+                                <IconButton
+                                    className={"query-option"}
+                                    value={QUERY_OPTION.IS_REGEX}
+                                >
+                                    .*
+                                </IconButton>
+                            </ToggleButtonGroup>
+                        }
+                        slotProps={{
+                            textarea: {className: "query-input-box-textarea"},
+                            endDecorator: {className: "query-input-box-end-decorator"},
+                        }}
+                        onChange={handleQueryInputChange}/>
+                    <LinearProgress
+                        determinate={true}
+                        sx={{"--LinearProgress-progressRadius": 0}}
+                        thickness={4}
+                        value={queryProgress * 100}
+                        color={1 === queryProgress ?
```

Comment:
Following https://github.com/y-scope/yscope-log-viewer/pull/107/files#r1837632878 , we shall extract a const to avoid magic numbers:
```suggestion
                        color={QUERY_PROGRESS_VALUE_MAX === queryProgress ?
```